File: src/j2ee/com.ibm.sbt.web/WebContent/js/sdk/sbt/base/DataHandler.js
/*
 * © Copyright IBM Corp. 2012,2013
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); 
 * you may not use this file except in compliance with the License. 
 * You may obtain a copy of the License at:
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, 
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
 * implied. See the License for the specific language governing 
 * permissions and limitations under the License.
 */

/**
 * Social Business Toolkit SDK. Helpers for the base capabilities of data
 * handlers.
 * 
 * @module sbt.base.DataHandler
 */
define([ "../declare", "../lang" ], function(declare,lang) {

    /**
     * DataHandler class
     * 
     * @class DataHandler
     * @namespace sbt.base
     */
    var DataHandler = declare(null, {

        /**
         * Data type for this DataHandler
         */
        dataType : null,

        /**
         * Data for this DataHandler
         */
        data : null,

        /**
         * @constructor
         * @param {Object}
         *            args Arguments for this data handler.
         */
        constructor : function(args) {
            lang.mixin(this, args);
        },

        /**
         * @method getAsString
         * @param data
         * @returns
         */
        getAsString : function(property) {
            return null;
        },

        /**
         * @method getAsNumber
         * @returns
         */
        getAsNumber : function(property) {
            return null;
        },

        /**
         * @method getAsDate
         * @returns
         */
        getAsDate : function(property) {
            return null;
        },

        /**
         * @method getAsBoolean
         * @returns
         */
        getAsBoolean : function(property) {
            return null;
        },

        /**
         * @method getAsArray
         * @returns
         */
        getAsArray : function(property) {
            return null;
        },

        /**
         * @method getEntityId
         * @returns
         */
        getEntityId : function(data) {
            return null;
        },

        /**
         * @param parent
         * @returns
         */
        getEntityData : function(parent) {
            return data;
        },

        /**
         * @method getSummary
         * @returns
         */
        getSummary : function() {
            return null;
        },

        /**
         * @method getEntitiesDataArray
         * @returns {Array}
         */
        getEntitiesDataArray : function() {
            return [];
        }

    });
    return DataHandler;
});

File: sdk/com.ibm.sbt.core/src/main/java/com/ibm/sbt/services/client/base/datahandlers/DataHandler.java
/*
 * © Copyright IBM Corp. 2013
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); 
 * you may not use this file except in compliance with the License. 
 * You may obtain a copy of the License at:
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, 
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
 * implied. See the License for the specific language governing 
 * permissions and limitations under the License.
 */

package com.ibm.sbt.services.client.base.datahandlers;

import java.util.Date;
import java.util.List;
import java.util.Locale;

/**
 * This abstract class defines an interface to decouple the data format used so that it can be be replaced easily for another format. 
 * @author Carlos Manias
 *
 * @param <DataType>
 */
public interface DataHandler<DataType> {
	
	public String getAsString(FieldEntry field);
	public String getAsString(String field);
	public Date getAsDate(FieldEntry fieldName, final Locale locale) throws DataHandlerException;
	public Date getAsDate(FieldEntry field) throws DataHandlerException;
	public Date getAsDate(String field) throws DataHandlerException;
	public Date getAsDate(String field, final Locale locale) throws DataHandlerException;
	public int getAsInt(FieldEntry fieldName);
	public int getAsInt(String fieldName);
	public float getAsFloat(FieldEntry fieldName);
	public float getAsFloat(String fieldName);
	public boolean getAsBoolean(FieldEntry fieldName);
	public boolean getAsBoolean(String fieldName);
	public List<DataType> getEntries(FieldEntry path);
	public List<DataType> getEntries(String path);
	public DataType getEntry(FieldEntry path);
	public DataType getEntry(String path);
	public void setData(final Object data);
	public Object getData();
	public Long getAsLong(String fieldName);
	public Long getAsLong(FieldEntry fieldName);
	public String[] getAsArray(String fieldName);
	public String[] getAsArray(FieldEntry fieldName);
	
}

